lib/mocha/parameter_matchers/positional_or_keyword_hash.rb

```diff
@@ -0,0 +1,33 @@
+require 'mocha/configuration'
+require 'mocha/parameter_matchers/base'
+
+module Mocha
+  module ParameterMatchers
+    class PositionalOrKeywordHash < Base
```

Comment:
Open to a better name. Note that naming this `Hash` as well may cause some unexpected constant lookup failures e.g. in `HasEntry#parse_option`

Reply:
> Open to a better name. Note that naming this `Hash` caused some constant lookup failures due to namespacing e.g. in `HasEntry#parse_option`, so I thought it'd be safer to name this something else (instead of checking and changing all `Hash`es in the codebase to explicit reference the top-level `Hash`).

That makes sense to me and I can't immediately think of a better name. 👍 

> Also, should this be `# @private`?

Yes, I think the whole class could be marked as private for documentation purposes since I think it's only used internally. That way you could remove the method-level YARD annotations.